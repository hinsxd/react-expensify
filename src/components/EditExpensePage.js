import React, { Component } from 'react';
import ExpenseForm from './ExpenseForm';
import { connect } from 'react-redux';
import { editExpense, removeExpense } from '../actions/expenses';

export class EditExpensePage extends Component {
  onSubmit = expense => {
    this.props.editExpense(this.props.expense.id, expense);
    this.props.history.push('/');
  };
  onRemove = () => {
    this.props.removeExpense({ id: this.props.expense.id });
    this.props.history.push('/');
  };
  render() {
    return (
      <div>
        <h1>Edit Expense</h1>
        <p>Editing {this.props.expense.id}</p>
        <ExpenseForm expense={this.props.expense} onSubmit={this.onSubmit} />
        <button onClick={this.onRemove}>Remove</button>
      </div>
    );
  }
}

const mapStateToObject = (state, props) => ({
  expense: state.expenses.find(expense => expense.id === props.match.params.id)
});
const mapDispatchToProps = (dispatch, props) => ({
  editExpense: expense => dispatch(editExpense(expense.id, expense)),
  removeExpense: data => dispatch(removeExpense(data))
});

export default connect(
  mapStateToObject,
  mapDispatchToProps
)(EditExpensePage);
