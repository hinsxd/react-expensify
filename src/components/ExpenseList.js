import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';
export const ExpenseList = props => (
  <div>
    {props.expenses.length === 0 ? (
      <p>No Expense </p>
    ) : (
      props.expenses.map(expense => (
        <ExpenseListItem key={expense.id} {...expense} />
      ))
    )}
    {}
  </div>
);

const mapStateToProps = state => ({
  expenses: selectExpenses(state)
});

export default connect(mapStateToProps)(ExpenseList);
