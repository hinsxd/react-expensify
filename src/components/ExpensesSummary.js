import React from 'react';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral';
import { connect } from 'react-redux';
export const ExpensesSummary = ({ expensesCount, expensesTotal }) => {
  const expenseWord = expensesCount === 1 ? 'expense' : 'expenses';
  const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00');
  return (
    <div>
      <h1>
        Viewing {expensesCount} {expenseWord} totalling {formattedExpensesTotal}
      </h1>
    </div>
  );
};
const mapStateToProps = state => ({
  expensesTotal: selectExpensesTotal(selectExpenses(state)),
  expensesCount: selectExpenses(state).length
});

export default connect(mapStateToProps)(ExpensesSummary);
