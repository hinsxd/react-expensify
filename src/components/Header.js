import React from 'react';
import { NavLink } from 'react-router-dom';
const Header = () => (
  <header>
    <h1>Expensify</h1>
    <NavLink to="/" activeClassName="is-acrtive" exact={true}>
      Home
    </NavLink>
    <NavLink to="/create" activeClassName="is-acrtive">
      Create Expense
    </NavLink>
  </header>
);

export default Header;
