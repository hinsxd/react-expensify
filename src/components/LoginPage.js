import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogin, startLogout } from '../actions/auth';

export const LoginPage = ({ startLogin, startLogout }) => (
  <div>
    <button onClick={startLogin}>Login</button>
  </div>
);

const mapDispatchToProps = dispatch => ({
  startLogin: () => dispatch(startLogin()),
  startLogout: () => dispatch(startLogout())
});

export default connect(
  undefined,
  mapDispatchToProps
)(LoginPage);
