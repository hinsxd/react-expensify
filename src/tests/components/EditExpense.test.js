import React from 'react';
import { EditExpensePage } from '../../components/EditExpensePage';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';

let editExpense, removeExpense, history, wrapper, expense;

beforeEach(() => {
  editExpense = jest.fn();
  removeExpense = jest.fn();
  history = { push: jest.fn() };
  expense = expenses[1];
  wrapper = shallow(
    <EditExpensePage
      expense={expense}
      editExpense={editExpense}
      removeExpense={removeExpense}
      history={history}
    />
  );
});

test('should render editExpensePage correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should handle editExpense', () => {
  wrapper.find('ExpenseForm').prop('onSubmit')(expense);
  expect(history.push).toHaveBeenLastCalledWith('/');
  expect(editExpense).toHaveBeenLastCalledWith(expense.id, expense);
});

test('should handle removeExpense', () => {
  wrapper.find('button').simulate('click');
  expect(history.push).toHaveBeenLastCalledWith('/');
  expect(removeExpense).toHaveBeenLastCalledWith({ id: expense.id });
});
