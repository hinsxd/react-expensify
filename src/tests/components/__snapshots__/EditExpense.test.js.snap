// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render editExpensePage correctly 1`] = `
<div>
  <h1>
    Edit Expense
  </h1>
  <p>
    Editing 
    1
  </p>
  <ExpenseForm
    expense={
      Object {
        "amount": 3,
        "createdAt": "1969-12-28T00:00:00.000Z",
        "description": "gum",
        "id": "1",
        "note": "some note",
      }
    }
    onSubmit={[Function]}
  />
  <button
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
